docs/howtos/locally-published-spm-in-firefox.md

```diff
@@ -0,0 +1,105 @@
+It's important to note the automated flow runs through those steps in a script, so if possible use the script as it's a tedious manual process. However, if the script is failing or you would like to run the manual process for any other reason follow the following steps.
+
+### Building the xcframework
+To build the [xcframework](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages) do the following:
+1. In your local checkout of `application-services`, navigate to [`megazords/ios-rust/`](https://github.com/mozilla/application-services/tree/main/megazords/ios-rust)
+1. Run the `build-xcframework.sh` script: 
+```bash
+$ ./build-xcframework.sh
+```
+This will produce a file name `MozillaRustComponents.xcframework.zip` that contains the following, built for all our target iOS platforms.
+- The compiled Rust code for all the crates listed in `Cargo.toml` as a static library
+- The C header files and [Swift module maps](https://clang.llvm.org/docs/Modules.html) for the components
+
+## Include the xcframework in a local checkout of `rust-components-swift`
+After you generated the `MozillaRustComponents.xcframework.zip` in the previous step, do the following to include it in a local checkout of `rust-components-swift`:
+1. Unzip the `MozillaRustComponents.xcframework.zip` into the `rust-components-swift` repository: (Assuming you are in the root of the `rust-components-swift` directory and `application-services` is a neighbor directory)
+    ```sh
+     unzip  ../application-services/megazords/ios-rust/MozillaRustComponents.xcframework.zip -d .
+    ```
+1. Change the `Package.swift`'s reference to the xcframework to point to unzipped folder. You can do this by uncommenting the following line:
+    ```swift
+        path: "./MozillaRustComponents.xcframework"
+    ```
+    and commenting out the following lines:
+    ```swift
+        url: url,
+        checksum: checksum,
+    ```
+
+## Run the make-tag script with a local checkout of application services
+For this step, run the following script from inside the `rust-components-swift` repository (assuming that `application-services` is a neighboring directory to `rust-components-swift`). Change the `X.Y.Z` to be a valid semver version, for example: `0.0.101`
+```sh
+./make_tag.sh -l ../application-services X.Y.Z
+```
+Once that is done, your local checkout will now have a git tag `X.Y.Z` that can be pointed to by Xcode
+
+## Include the local checkout of `rust-components-swift` in `firefox-ios`
+This is the final step to include your local changes into `firefox-ios`. Do the following steps:
+1. Open `Client.xcodeproj` in Xcode
+1. Navigate to the Swift Packages in Xcode:
+![Screenshot of where to find the setting for Client](./img/xcode-client-package-settings.png)
+1. Remove the dependency on `rust-components-swift` as listed on Xcode, you can click the dependency then click the `-`
+1. Add a new swift package by clicking the `+`:
```

Comment:
I was surprised to find my xcode already had a file:// reference to my local checkout, but then noticed your screenshot does too - but it does point to main. Is there a reason we can't just change that to the new version?

Also, in "Add to Project" I don't have a "Blockzilla", just "Client"

This all seemed to work - but the build failed with this - so I guess I didn't clobber first?

![image](https://user-images.githubusercontent.com/83924/168975435-aec07f50-6448-46e3-acc2-2bc229aa616c.png)

Reply:
> I was surprised to find my xcode already had a file:// reference to my local checkout, but then noticed your screenshot does too

I'm guessing it's because you used the script before 😬 The script does everything the manual steps do

> This all seemed to work - but the build failed with this - so I guess I didn't clobber first?

Yeah, probably re-running the `bootstrap.sh` they have on firefox-ios would fix that